basic_system/src/system_implementation/flat_storage_model/account_cache.rs

```diff
@@ -547,6 +551,11 @@ where
                 }
             })?,
             code_version: Maybe::construct(|| full_data.versioning_data.code_version()),
+            is_delegated: Maybe::construct(|| {
+                // Only EVM accounts can be delegated
+                full_data.versioning_data.ee_version() == ExecutionEnvironmentType::EVM_EE_BYTE
+                    && full_data.versioning_data.is_delegated()
```

Comment:
Should we have an assert here?

Reply:
Good point, this is an internal invariant. Done